registration/backends.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015 OpenCraft <xavier@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Authentication backends
+"""
+
+# Imports #####################################################################
+
+from django.contrib.auth import get_user_model
+from django.contrib.auth import backends
+from django.db.models import Q
+
+
+class ModelBackend(backends.ModelBackend):
+    """
+    Extends the default ModelBackend to fetch users by email address as well as
+    by username.
+    """
+    def authenticate(self, username=None, password=None):
+        """
+        This is mostly copied from the default ModelBackend. Attempts to fetch
+        users by username or email address.
```

Comment:
@omarkhan could be helpful to mention what's different; otherwise it is not clear why can't we use default ModelBackend (actually, can't we?)